### PR TITLE
Update icu_properties.yaml for version 1.5.1

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
Update license of version and source location information

### CODE REVIEW

**Comment by SteveL-MSFT** [comment]
The license matches https://spdx.org/licenses/Unicode-3.0.html

**Comment by capfei** [comment]
@SteveL-MSFT Thank you for the contribution!

Can you update the source location url to `https://github.com/unicode-org/icu4x/commit/e1193d0f5d343317857570e9c4e8a03c87875d8b`?

We use `https://github.com/{owner}/{repo}/commits/{commit_sha}` for [GitHub source locations](https://github.com/clearlydefined/clearlydefined/blob/main/docs/curation/curation-guidelines.md#supported-source-locations).

The validation is out of date, which is why it is failing. I have been manually updating and merging any that use newer identifiers until it gets fixed (hopefully soon).

File: curations/crate/cratesio/-/icu_properties.yaml [hunk 1]
```diff
@@ -6,3 +6,14 @@ revisions:
...
+    licensed:
+      declared: Unicode-3.0
+    described:
+      sourceLocation:
+        name: icu4x
+        namespace: unicode-org
+        provider: github
+        revision: e1193d0f5d343317857570e9c4e8a03c87875d8b
+        type: git
+        url: 'https://github.com/unicode-org/icu4x/tree/ind/icu_properties%401.5.1'
```
Comment (SteveL-MSFT) [inline curations/crate/cratesio/-/icu_properties.yaml:19]:
```suggestion
        url: 'https://github.com/unicode-org/icu4x/commit/e1193d0f5d343317857570e9c4e8a03c87875d8b'
```

**Comment by SteveL-MSFT** [comment]
@capfei Done. Thanks!

**Pull request merged**